core/src/main/java/org/elasticsearch/index/query/FieldMaskingSpanQueryBuilder.java

```diff
@@ -72,6 +112,72 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String fieldInQuery = this.fieldName;
+        MappedFieldType fieldType = parseContext.fieldMapper(fieldName);
+        if (fieldType != null) {
+            fieldInQuery = fieldType.names().indexName();
+        }
+        SpanQuery innerQuery = (SpanQuery) ((QueryBuilder) this.queryBuilder).toQuery(parseContext);
```

Comment:
Okay, this is getting a bit complicted now:
I re-checked to see now that back in may you made SpanQueryBuilder an abstract class extending QueryBuilder on master, but during the refactoring of the SpanTermQuery (https://github.com/elastic/elasticsearch/pull/11005) this was changed back to be an interface on our feature branch. I tried to change it back to be an abstract class like on master, but now I have trouble with the inheritance hierarchy of SpanTermQuery. Here we have:
* `SpanTermQueryBuilder extends BaseTermQueryBuilder<SpanTermQueryBuilder>`
* `TermQueryBuilder extends BaseTermQueryBuilder<TermQueryBuilder>`
* `abstract class BaseTermQueryBuilder<QB extends BaseTermQueryBuilder<QB>> extends QueryBuilder<QB>`

I'm having trouble now fitting the SpanQueryBuilder in here. If I merge it in the way it is on master it sits between all the other Span*Q.B. and QueryBuilder, but for the SpanTermQB I cannot easily make it extend BaseTermQueryBuilder. Maybe I'm missing something here. Maybe this was the reason it was changed back in #11005 to be a marker interface?
 

```